crates/credentials/src/vc.rs

```diff
@@ -6,16 +6,52 @@ use jwt::{
 };
 use serde::{Deserialize, Serialize};
 use std::{collections::HashMap, sync::Arc};
+use chrono::Utc;
 
 const BASE_CONTEXT: &str = "https://www.w3.org/2018/credentials/v1";
 const BASE_TYPE: &str = "VerifiableCredential";
 
 #[derive(thiserror::Error, Debug)]
 pub enum CredentialError {
+    // JWT specific errors
     #[error(transparent)]
     JwtError(#[from] JwtError),
     #[error(transparent)]
     JwsError(#[from] JwsError),
+
+    // Credential-specific validation errors
+    #[error("Expiration date mismatch")]
+    ExpirationDateMismatch,
+    #[error("Issuer mismatch")]
+    IssuerMismatch,
+    #[error("Issuance date is in the future")]
+    IssuanceDateInFuture,
+    #[error("Issuance date mismatch")]
+    IssuanceDateMismatch,
+    #[error("ID mismatch")]
+    IdMismatch,
+    #[error("Subject mismatch")]
+    SubjectMismatch,
+    #[error("Missing issuer")]
+    MissingIssuer,
+    #[error("Misconfigured expiration date: {0}")]
+    MisconfiguredExpirationDate(String),
+
+    // Data model validation error
+    #[error("Validation error: {0}")]
```

Comment:
FYI the reason this is failing is because you need to add the `thiserror::Error` to your `CredentialDataModelValidationError` enum below, and then attribute each enum value with the `#[error()]` enum

I just opened a ticket relevant to this, again not proposing a change with your code, just raising awareness https://github.com/TBD54566975/web5-rs/issues/184

Reply:
also I would recommend this instead, because then you don't have to make the call to `.map_err` in your validate call below 

```rust
    #[error(transparent)]
    CredentialDataModelValidationError(#[from] CredentialDataModelValidationError),
```